clusterloader2/GETTING_STARTED.md

```diff
@@ -0,0 +1,227 @@
+# Once we started measurements and created deployment, now we can wait for this deployment to be running.
+# Also, we can measure pod startup latency
+- name: Wait for pods to be running
+  measurements:
+  - Identifier: WaitForControlledPodsRunning
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: gather
+  - Identifier: PodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: gather
+```
+Now, we need to prepare template for our deployment to be created. All values in brackets will be filled from template parameters:
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+# Name is parameter that will be autogenerated by ClusterLoader
+  name: {{.Name}}
+  labels:
+    group: test-pod
+spec:
+  replicas: {{.Replicas}}
+  selector:
+    matchLabels:
+      name: {{.Name}}
+  template:
+    metadata:
+      labels:
+        name: {{.Name}}
+        group: test-pod
+    spec:
+      containers:
+      - image: k8s.gcr.io/pause:3.1
+        imagePullPolicy: IfNotPresent
+        name: {{.Name}}
+        resources:
+          requests:
+            cpu: 1m
+            memory: 10M
+```
+## Execute test
+Before running test, make sure that kubeconfig current context points to kind cluster:
+```bash
+kubectl config current-context
+kind-test-cluster
+```
+To execute test, run:
+```bash
+cd clusterloader2/
+go run cmd/clusterloader.go --testconfig=config.yaml --provider=kind --kubeconfig=${HOME}/.kube/config --v=2
+```
+
+At the end of clusterloader output you should see pod startup latency:
+```json
+    {
+      "data": {
+        "Perc50": 1252.347301,
+        "Perc90": 1252.347301,
+        "Perc99": 1252.347301
+      },
+      "unit": "ms",
+      "labels": {
+        "Metric": "schedule_to_watch"
+      }
+    },
+```
+"schedule_to_watch"  measures time since pod was scheduled until it was observed via watch as running.
+You should also see that test succeeded:
+```
+--------------------------------------------------------------------------------
+Test Finished
+Test: ./config.yaml
+Status: Success
+--------------------------------------------------------------------------------
+```
+
+## Delete kind cluster
+To delete kind cluster, run:
+```bash
+kind delete cluster --name test-cluster
+```
+
+## Running 100-node scale test
+
+Here you can find general purpose [Load test]. This test is release-blocking test we use to evaluate scalability of kubernetes.
+It consists of 3 main phases:
+- Creating objects
+- Scaling objects in range (-50%, 50%)
+- Deleting objects
+
+It can be used to test clusters starting from 100 nodes up to 5k nodes.
+Load test will create, roughly 30 * nodes objects, divided into:
+- 1/4 of pods in big deployments (250 pods each)
+- 1/4 of pods in medium deployments (30 pods each)
+- 1/2 of pods in small deployments (5 pods each)
+- 1 small statefulset (5 pods)
+- 1 medium statefulset (30 pods)
+- 1 small job (5 pods)
+- 1 medium job (30 pods)
+- 1 big job (250 pods)
```

Comment:
This will get out of data (and I don't think such detailed data matters).

Let's just mention that:
- we create: deployments, statefulsets, jobs and daemonset
- there are small (5), medium (30) and big (250) versions of them

The exact numbers don't matter much and can become easily outdated.

Let's also mention services, secrets and configmaps.

Reply:
Done.